llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -3248,19 +3248,18 @@ InstructionCost AArch64TTIImpl::getVectorInstrCostHelper(
     // Check if the extractelement user is scalar fmul.
     auto IsUserFMulScalarTy = [](const Value *EEUser) {
       // Check if the user is scalar fmul.
-      const auto *BO = dyn_cast_if_present<BinaryOperator>(EEUser);
+      const auto *BO = dyn_cast<BinaryOperator>(EEUser);
```

Comment:
EEUser can be null and hence, dont change this

Reply:
This function is called from users() in two cases. Can you explain why it can be null? Do you have a test case that shows it?

Reply:
sorry, my bad. You are right. (if_present was required in one of the earlier revisions of the original patch and that lasted till end)